component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1658,4 +1669,120 @@ private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) th
+        } catch (IdentityEventException e) {
+            String errorMsg = "An error occurred while triggering post event in SMS OTP generation flow. " + e.getMessage();
+            throw new AuthenticationFailedException(errorMsg, e);
+        }
+    }
+
+    /**
+     * Trigger event after validating SMS OTP.
+     *
+     * @param request HttpServletRequest.
+     * @param context Authentication context.
+     * @throws AuthenticationFailedException Exception on authentication failure.
+     */
+    private void publishPostSMSOTPValidatedEvent(HttpServletRequest request,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        AuthenticatedUser authenticatedUser = (AuthenticatedUser) context.getProperty(SMSOTPConstants
+                .AUTHENTICATED_USER);
+        eventProperties.put(IdentityEventConstants.EventProperty.CORRELATION_ID, context.getCallerSessionKey());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_NAME, authenticatedUser.getUserName());
+        eventProperties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, context.getTenantDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, authenticatedUser
+                .getUserStoreDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.APPLICATION_NAME, context.getServiceProviderName());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_AGENT, request.getHeader(
+                SMSOTPConstants.USER_AGENT));
+
+        eventProperties.put(IdentityEventConstants.EventProperty.CLIENT_IP, IdentityUtil.getClientIpAddress(request));
+        eventProperties.put(IdentityEventConstants.EventProperty.GENERATED_OTP, context.getProperty(
+                SMSOTPConstants.OTP_TOKEN));
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_INPUT_OTP, request.getParameter(
+                SMSOTPConstants.CODE));
+        eventProperties.put(IdentityEventConstants.EventProperty.OTP_USED_TIME, System.currentTimeMillis());
+
+        long otpGeneratedTime = (long) context.getProperty(SMSOTPConstants.SENT_OTP_TOKEN_TIME);
+        eventProperties.put(IdentityEventConstants.EventProperty.OTP_GENERATED_TIME,
+                otpGeneratedTime);
+
+        String otpExpiryTime = SMSOTPUtils.getTokenExpiryTime(context);
+        if (StringUtils.isNotEmpty(otpExpiryTime)) {
+            long expiryTime = otpGeneratedTime + Long.parseLong(otpExpiryTime);
+            eventProperties.put(IdentityEventConstants.EventProperty.OTP_EXPIRY_TIME, expiryTime);
+        }
+
+        String status;
+        if (context.getProperty(SMSOTPConstants.TOKEN_EXPIRED) != null && context.getProperty(
+                SMSOTPConstants.TOKEN_EXPIRED) == SMSOTPConstants.TOKEN_EXPIRED_VALUE) {
```

Comment:
Let's use the `equals` function. 

```
SMSOTPConstants.TOKEN_EXPIRED_VALUE.equals(context.getProperty(SMSOTPConstants.TOKEN_EXPIRED))
```

Then we do not need the null check.

Reply:
Fixed in 2a8bda4c65d9afc43f175c6556e374838ebd3e60